### ISSUE TITLE
Initial skeleton files copy overwrites existing shared files

### ISSUE THREAD

**Issue by PVince81** [owncloud/core#16450]
### Steps
1. Checkout ownCloud from git (or put a file "welcome.txt" in core/skeleton)
2. Setup with admin "admin"
3. Login as "admin"
4. Edit the file "welcome.txt" and replace the contents with the string "MODIFIED"
5. Create a user "user1"
6. Share "welcome.txt" with "user1"
7. Login as "user1"
8. Check the contents of "welcome.txt"
9. Login as "admin"
10. Check the contents of "welcome.txt"
### Expected

At step 7, I should see two files "welcome.txt" (the shared one) and "welcome (2).txt" the one from the skeleton. At step 10, the file "welcome.txt" from "admin" still contains "MODIFIED".
### Actual result

At step 7, the file "welcome.txt" is the one from the skeleton.
At step 10, the file from "welcome.txt" contains again the text from the skeleton instead of "MODIFIED".
What happened is that the skeleton logic overwrote the **contents** of the existing shared file, which then overwrote what the admin user sees.
### Versions

Observed in 8.1 on master (12aef311154f12050c4e33492dac1fb1c928060d)

**Comment by oparoz** [comment]
Actually, it would be great if the skeleton copying routine could simply stop if there is existing content.

**Comment by PVince81** [comment]
True. Maybe have a user flag in the database that says that the user has already been setup.

But in this specific use case the user has not be setup yet, but people might still be able to share stuff with them before their first login, which is a valid use case.

**Comment by nickvergessen** [comment]
Sounds like a case for smashbox?

**Comment by PVince81** [comment]
Yes, would be good to cover that.

**Comment by nickvergessen** [comment]
Created https://github.com/owncloud/smashbox/issues/87

**Comment by PVince81** [comment]
Also causes trouble with folders: https://github.com/owncloud/core/issues/16899

**Comment by fuller42** [comment]
This means i have to wait for a new user to login before i can put him in groups (with shares) or otherwise he wont get the skeleton contents.

**Comment by nickvergessen** [comment]
Or you name your share different to the skeleton ;)

**Comment by fuller42** [comment]
> Or you name your share different to the skeleton ;) -nickvergessen
nope, has nothing to do with the overlapping name thing.
never logged on + has shares = no skeleton

**Comment by davitol** [comment]
Initial skeleton files are also overwritten if admin runs transfer-ownership command against a never logged account

**Comment by rullzer** [comment]
Yes most likely.
@DeepDiver1975 we should make sure the 'login' the new user fully once before we run the migration.

**Comment by ownclouders** [comment]
Hey, this issue has been closed because the label `status/STALE` is set and there were no updates for 7 days. Feel free to reopen this issue if you deem it appropriate.

(This is an automated comment from [GitMate.io](https://gitmate.io).)

**Comment by ownclouders** [comment]
Hey, this issue has been closed because the label `status/STALE` is set and there were no updates for 7 days. Feel free to reopen this issue if you deem it appropriate.

(This is an automated comment from [GitMate.io](https://gitmate.io).)
